bin/schema_generator.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+# A script for generating schemas that are deployed into the pipeline. This
+# script handles preprocessing, filtering, and transpilation of schemas as part
+# of a pre-deployment scheme. The resulting schemas are pushed to a branch of
+# mozilla-pipeline-schemas.
+#
+# Environment variables:
+#   MPS_SSH_KEY_BASE64: A base64-encoded ssh secret key with permissions to push
+#                       to mozilla-pipeline-schemas
+#
+# Example usage:
+#   export MPS_SSH_KEY_BASE64=$(cat ~/.ssh/id_rsa | base64)
+#   make build && make run
+#
+# TODO: Update schema mapping for validation
+# TODO: Handle overwriting glean schemas
+# TODO: Include Main Ping from schema generation
+# TODO: What the heck to do with pioneer-study, a non-nested namespace
+
+MPS_BRANCH_SOURCE="dev"
+MPS_BRANCH_WORKING="generated-schemas-dev"
+MPS_BRANCH_PUBLISH="generated-schemas"
+MPS_SCHEMAS_DIR="schemas"
+
+BASE_DIR="/app"
+
+
+function setup_git_ssh() {
+    # Configure the container for pushing to github
+
+    if [[ -z "$MPS_SSH_KEY_BASE64" ]]; then
+        echo "Missing secret key" 1>&2
+        exit 1
+    fi
+
+    git config --global user.name "Generated Schema Creator"
+    git config --global user.email "dataops+pipeline-schemas@mozilla.com"
+
+    mkdir -p /app/.ssh
+
+    echo "$MPS_SSH_KEY_BASE64" | base64 --decode > /app/.ssh/id_ed25519
+    ssh-keyscan github.com > /app/.ssh/known_hosts # Makes the future git-push non-interactive
+
+    chown -R "$(id -u):$(id -g)" "$HOME/.ssh"
+    chmod 700 "$HOME/.ssh"
+    chmod 700 "$HOME/.ssh/id_ed25519"
+}
+
+function setup_dependencies() {
+    # Installs mozilla-schema-generator in a virtual environment
+
+    virtualenv msg-venv
+    # shellcheck disable=SC1091
+    source msg-venv/bin/activate
+    pip install -e ./mozilla-schema-generator
+}
+
+function clone_and_configure_mps() {
+    # Checkout mozilla-pipeline-schemas and changes directory to prepare for
+    # schema generation.
+
+    rm -rf mozilla-pipeline-schemas
```

Comment:
Generally I would prefer something like `[[ -d mozilla-pipeline-schemas ]] && rm -rf mozilla-pipeline-schemas` but this is fine.